addons/mail/static/src/messaging/entity/composer/composer.js

```diff
@@ -50,6 +84,27 @@ function ComposerFactory({ Entity }) {
             });
         }
 
+        /**
+         * @param mentionKeyword
+         */
+        insertSuggestedPartner(mentionKeyword) {
+            const selection = this.suggestedPartners.find(suggestedPartner => suggestedPartner == this.activePartner);
+            const cursorPosition = this.textInputCursorStart;
+            const textLeft = this.textInputContent.substring(0, cursorPosition - (mentionKeyword.length));
+            const textRight = this.textInputContent.substring(cursorPosition, this.textInputContent.length);
+            const insert = selection.name.replace(/ /g, '\u00a0');
+            this.update({
+                suggestedPartners: [['unlink-all']],
+                textInputContent: textLeft + insert + ' ' + textRight,
+                textInputCursorStart: textLeft.length + selection.name.length + 2,
+                textInputCursorEnd: textLeft.length + selection.name.length + 2,
+            });
```

Comment:
I think `activePartner` should be "consumed" (unlinked) here when it is used, there is no point to keep the `activePartner` when `suggestedPartners` is empty. (maybe `activePartner` should become a compute with `suggestedPartners` as dependency, that computes itself to unlink if its not in the suggested anymore, but that keeps its current value otherwise). cc @alexkuhn